nncf/tensorflow/graph/model_transformer.py

```diff
@@ -105,23 +132,27 @@ def _get_layer(self, layer_name):
 
         return None
 
-    def _find_layer_config(self, layer_name):
+    def _find_layer_config(self, layer_name: str):
         for idx, layer in enumerate(self._model_config['layers']):
             layer_name_ = layer['name'] if is_functional_model(self._model) \
                 else layer['config']['name']
             if layer_name_ == layer_name:
                 return idx, layer
         return None, None
 
-    def _update_layer_mapping(self, src_layer_name, dst_layer_name):
+    def _update_layer_mapping(self, src_layer_name: str, dst_layer_name: str):
         if src_layer_name in self._name_mapping.values():
             for orig_layer_name in self._name_mapping:
                 if self._name_mapping[orig_layer_name] == src_layer_name:
                     self._name_mapping[orig_layer_name] = dst_layer_name
         else:
             self._name_mapping[src_layer_name] = dst_layer_name
 
-    def _apply_transformation(self, transformation):
+    def _apply_transformation(self, transformation: TransformationCommand):
+        target_point = transformation.target_point
+        if transformation.type != TransformationType.REMOVE and target_point not in self._allowed_target_points:
+            raise RuntimeError(f"Target point not allowed: {str(target_point)}")
```

Comment:
The compression algos should only attempt to modify the target points that are actually available. Either the available target points should be explicitly passed to each algo (requires API changes), or there should be an implicit synchronization between what each backend should support in terms of target points if it wishes to use a particular algo, and with what kind of target points this particular algo generates. I implemented the second scenario for the time being, and the check here is absolutely necessary so that any desync between the converter/transformer capabilities and the algo requirements is caught early.